qr-code-auth/README.md

```diff
@@ -0,0 +1,277 @@
+# Collection - QrCodeAuth
+
+## API Endpoints
+
+|                                       Route                                        |                                         Description                                          |
+| :--------------------------------------------------------------------------------: | :------------------------------------------------------------------------------------------: |
+|                      [POST /auth/qr-code-auth](#post-authqr-code-auth)                      |                            Creates a new authentication document.                            |
+|                       [GET /auth/qr-code-auth](#get-authqr-code-auth)                       |           Retrieves all the authentication document belonging to a specified user            |
+| [PATCH /auth/qr-code-auth](#patch-authqr-code-authauthorization_statusauthorization_status) | Updates the authorization_status field of an existing qr-code-auth document for the specified user. |
+
+## POST /auth/qr-code-auth
+
+    Creates a new qr-code-auth document.
+
+- **Params**  
+  None
+- **Query**
+  None
+- **Body**
+  - Attributes
+    - **user_id** (required, string): Specifies the user id for the document.
+    - **device_info** (required, string): Specifies the device information associated with the authentication.
+- **Headers**
+  - Content-Type: application/json
+- **Cookie**
+  None
+- **Success Response:**
+  - **Code:** 201
+    - **Content:**
+      ```json
+      {
+        "message": "String",
+        "data": {
+          "user_id": "String",
+          "device_info": "String",
+          "authorization_status": "String",
+          "access_token": "String"
+        }
+      }
+      ```
+- **Error Response:**
+
+  - **Code:** 400
+
+    - **Content:**
+      ```json
+      {
+        "message": "Bad Request"
+      }
+      ```
+
+  - **Code:** 409
+
+    - **Content:**
+      ```json
+      {
+        "message": "The authentication document has already been created"
+      }
+      ```
+
+  - **Code:** 500
+    - **Content:**
+      ```json
+      {
+        "message": "The server has encountered an unexpected error. Please contact the administrator for more information."
+      }
+      ```
+
+- **Example for user authentication document creation request:**
+  POST /auth/qr-code-auth<br/>
+  Content-Type: application/json<br/>
```

Comment:
Here anyone can make a post call with some other's userId and add data
Will it create security concerns?

Reply:
Yes, you're right. This could pose a security risk. @shreya-mishra  @sakshambhatt  as we talked and removed the cookie, Prakash has brought up a valid point. what do you think @shreya-mishra  @sakshambhatt ?

Reply:
Indeed they can. But it will not be a security issue because the entire system works on permission. 
Let us take an example. Suppose person X opens my site, signs in and opens qr code to sign in with mobile.
Now, they have our mobile app downloaded on their phone where they have not signed in yet. Behind them a 
malicious person named Y also sees this, and they scan the QR code from their phone before person X was able to.
This will send person Y's device info to the backend, and by clicking on button on my site, person X will be asked to authorize sign in on like following:
Apple - iPhone 7 has requested to sign in to your account on phone. Now, if person X doesn't have Apple - iPhone 7, they can simply deny the request.
So, now the issue is what happens in case that person X and Y have the same devices, or person Y knowing all this due to it being open source, sends request from their end with device info matching that of person X?
Maybe we should implement some form of OTP in the mobile which would be invisible to both X and Y, and send it in body along with device_info and user_id?
@prakashchoudhary07 @shreya-mishra

Reply:
then we can also send the device id which is going to be unique for every user.

Reply:
let's show both device-info and device-id for now, and later will change it to otp one.
cc: @prakashchoudhary07 
@heyrandhir 
@sakshambhatt

Reply:
I think this one is fine for now.